OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -58,6 +60,27 @@ public void WorldLoaded(World w, WorldRenderer wr)
 
 			foreach (var cell in Map.AllCells)
 				UpdateCell(cell);
+
+			// Build the sprite layer dictionary for rendering resources
+			// All resources that have the same palette must also share a sheet and blend mode
+			foreach (var r in Resources)
+			{
+				var layer = spriteLayers.GetOrAdd(r.Value.Palette, pal =>
+				{
+					var first = r.Value.Variants.First().Value.First();
```

Comment:
This isn't a problem in practice because the closure is evaluated immediately.  Fixed it to silence the warning, though.